### PR TITLE
Implemented article-link provider and custom ui

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | https://github.com/sulu/sulu/pull/2967 (merged) https://github.com/sulu/sulu/pull/2985 (bugfix for spacing)
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR introduces an `ArticleLinkProvider` which enables the editor to create internal links to articles.

### CODE REVIEW

**Comment by danrot** [comment]
@chirimoya Seems to work, can you have a look at the code? Don't really know that bundle.

File: Resources/public/js/components/ckeditor/link/article/main.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    'services/suluarticle/article-manager',
+    'text!/admin/api/articles/fields'
+], function(Config, Manager, fieldsResponse) {
+
+    'use strict';
+
+    // TODO merge with user settings.
+    // TODO column-options
+    var fields = JSON.parse(fieldsResponse),
+
```
Comment (alexander-schranz) [inline Resources/public/js/components/ckeditor/link/article/main.js]: remove this empty line

File: Resources/public/js/components/ckeditor/link/article/main.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        config = Config.get('sulu_article');
+
+    return {
+
```
Comment (alexander-schranz) [inline Resources/public/js/components/ckeditor/link/article/main.js]: remove this empty line

File: Resources/public/js/components/ckeditor/link/article/main.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                this.options.setHref(data.id, data.title, true);
+            }.bind(this));
+        },
+
+        render: function() {
+            var $container = $(this.templates.contentDatasource());
+            this.$el.append($container);
+
+            var type = '';
+            if (config.typeNames.length > 0) {
```
Comment (alexander-schranz) [inline Resources/public/js/components/ckeditor/link/article/main.js]: move to initialize or add a comment here to make clear why only the first type is used.

**Comment by alexander-schranz** [comment]
@wachterjohannes ACK

**Pull request merged**